heroku/cli
# refactor(pg-v5): Move command 'pg:backups:capture' to CLI

## Description

Moves the `pg:backups:capture` command to CLI, converting to OClif Core and TS.

Note that we're removing a hidden `--snapshot` flag from the current implementation, because the destination endpoint has been removed from Shogun and it doesn't work in our current CLI stable release. AFAIK it never worked, for context you can see: https://github.com/heroku/cli/pull/1209.

## Test instructions

You can test by pulling this branch, installing dependencies with `yarn` and building with `yarn build`. Then run the following commands:

```
./bin/run pg:backups:capture --help                                   # always test that help looks ok.
./bin/run pg:backups:capture -a test_app_without_pg_database          # shows error.
./bin/run pg:backups:capture -a test_app_with_mini_db                 # starts backup, doesn't show warning.
./bin/run pg:backups:capture -a test_app_with_premium_db              # starts backup, shows warning.
./bin/run pg:backups:capture -a test_app_with_db_attachment           # starts backup, shows HINT.
```

Repeat some commands using the flag `--verbose`. You should see log messages from the backup.
You can use a single test app with multiple DBs, but then you will need to pass the add-on name or URL connection config var as an argument to the command.

To add databases you can use ```heroku addons:create heroku-postgresql:<PLAN> -a test_app```. To create an attachment to a database billed to a different app, you can run ```heroku addons:attach ADDON_NAME -a attach_to_test_app```.

## SOC2 Compliance
[GUS Work Item](https://gus.lightning.force.com/a07EE00001b2EFgYAM)
